### PR TITLE
Remove dependency of Light on Application

### PR DESCRIPTION
**Pull request by mvaligursky**
- Light gets device in constructor and stores it locally to avoid dependency on the Application class
- as program library stringifies options that contain array of lights, which now stores device, device class implement toJSON to disable serialization
- these changes caused DefaultAnimBinder to fail type checks, and so it's dependency on Entity was removed as well, which removed 2 circular dependencies

### CODE REVIEW

**Review by ellthompson — APPROVED** [review]

File: src/graphics/graphics-device.js [hunk 1]
```diff
@@ -697,6 +697,11 @@ class GraphicsDevice extends EventHandler {
         }
     }
 
+    // don't stringify GraphicsDevice to JSON by JSON.stringify
+    toJSON(key) {
+        return undefined;
+    }
```
Comment (willeastcott) [inline src/graphics/graphics-device.js:703]: Why is this needed in this PR? Seems unrelated.

**Review by willeastcott — APPROVED** [review]
